functions/Enable-DbaHideInstance.ps1

```diff
@@ -0,0 +1,131 @@
+function Enable-DbaHideInstance {
+    <#
+    .SYNOPSIS
+        Enables Hide Instance of SQL Engine.
+
+    .DESCRIPTION
+        Enables Hide Instance of SQL Engine. Note that this requires access to the Windows Server, not the SQL instance itself.
+
+        This setting is found in Configuration Manager.
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances.
+
+    .PARAMETER Credential
+        Allows you to login to the computer (not SQL Server instance) using alternative Windows credentials
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: HideInstance, Security
+        Author: Gareth Newman (@gazeranco), ifexists.blog
+
+        Website: https://dbatools.io
+        Copyright: (c) 2019 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .EXAMPLE
+        PS C:\> Enable-DbaHideInstance
+
+        Enables Hide Instance of SQL Engine on the default (MSSQLSERVER) instance on localhost. Requires (and checks for) RunAs admin.
+
+    .EXAMPLE
+        PS C:\> Enable-DbaHideInstance -SqlInstance sql01\SQL2008R2SP2
+
+        Enables Hide Instance of SQL Engine for the SQL2008R2SP2 on sql01. Uses Windows Credentials to both connect and modify the registry.
+
+    .EXAMPLE
+        PS C:\> Enable-DbaHideInstance -SqlInstance sql01\SQL2008R2SP2 -WhatIf
+
+        Shows what would happen if the command were executed.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Low", DefaultParameterSetName = 'Default')]
+    param (
+        [Parameter(ValueFromPipeline)]
+        [DbaInstanceParameter[]]
+        $SqlInstance = $env:COMPUTERNAME,
+        [PSCredential]$Credential,
+        [switch]$EnableException
+    )
+    process {
+
+        foreach ($instance in $sqlinstance) {
+            Write-Message -Level VeryVerbose -Message "Processing $instance." -Target $instance
+            $null = Test-ElevationRequirement -ComputerName $instance -Continue
+
+            Write-Message -Level Verbose -Message "Resolving hostname."
+            $resolved = $null
+            $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
```

Comment:
Need to provide the `Credential` parameter and variable. As well, need to wrap this in a try/catch block and add the `-EnableException` parameter so the catch block is properly called. This ensures error output is captured properly by the logging system, in the event the underlying function is hosed up by some other change/commit.